### PR TITLE
DEV: Ensure stylesheet sourcemap URLs include __ws hostname param

### PR DESCRIPTION
**Pull request by davidtaylorhq**
Without this parameter, requests for sourcemaps on shared-CDN multisites will not be routed to the correct database, resulting in a 404.

The stylesheet content now depends on the site hostname, so the hostname has been added to the digest.

### CODE REVIEW

File: lib/stylesheet/manager/builder.rb [hunk 1]
```diff
@@ -86,19 +90,19 @@ def source_map_fullpath
...
+    "#{stylesheet_filename}.map?__ws=#{current_hostname}"
   end
 
   def stylesheet_fullpath_no_digest
     "#{cache_fullpath}/#{stylesheet_filename_no_digest}"
   end
 
-  def stylesheet_cdnpath(hostname)
-    "#{GlobalSetting.cdn_url}#{stylesheet_relpath}?__ws=#{hostname}"
+  def stylesheet_cdnpath
```
Comment (udan11) [inline lib/stylesheet/manager/builder.rb:100]: I do not think this is used anywhere, but the next method. Does it make sense to keep it around?

**Review by gschlager — APPROVED** [review]

**Review by udan11 — APPROVED** [review]

File: lib/stylesheet/manager/builder.rb [hunk 2]
```diff
@@ -86,19 +90,19 @@ def source_map_fullpath
   end
 
   def source_map_filename
-    "#{stylesheet_filename}.map"
+    "#{stylesheet_filename}.map?__ws=#{current_hostname}"
```
Comment (udan11) [inline lib/stylesheet/manager/builder.rb:93]: One curiosity: shouldn't `source_map_filename` just be `stylesheet_filename.sub(".scss", ".map")` or something like that? Why doesn't `stylesheet_filename` contain `__ws` too?

Comment (davidtaylorhq) [reply lib/stylesheet/manager/builder.rb]: Good point, this is quite confusing. I've just pushed a commit with some updated method names - does that look any better?

File: lib/stylesheet/manager/builder.rb (hunk 1 above)
Comment (davidtaylorhq) [reply lib/stylesheet/manager/builder.rb]: Removed :+1:

File: lib/stylesheet/manager/builder.rb [hunk 3]
```diff
@@ -36,7 +36,7 @@ def compile(opts = {})
          rtl: rtl,
          theme_id: theme&.id,
          theme_variables: theme&.scss_variables.to_s,
-         source_map_file: source_map_filename,
+         source_map_file: source_map_url_relative_from_stylesheet,
```
Comment (udan11) [inline lib/stylesheet/manager/builder.rb:39]: This is better, but I still find it a bit strange that we assign a relative URL to a filename. I don't know what "right" would look like, so I think you can leave it this way.

File: lib/stylesheet/manager/builder.rb (hunk 2 above)
Comment (udan11) [reply lib/stylesheet/manager/builder.rb]: This is better, but I still find it a bit strange that we assign a relative URL to a filename. I don't know what "right" would look like, so I think you can leave it this way.

**Review by udan11 — APPROVED** [review]

**Pull request merged**